docker-rancher/azuredeploy.json

```diff
@@ -0,0 +1,163 @@
+{
+  "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "adminUsername": {
+      "type": "string",
+      "metadata": {
+        "description": "Username for the Virtual Machine."
+      }
+    },
+    "adminPassword": {
+      "type": "securestring",
+      "metadata": {
+        "description": "Password for the Virtual Machine."
+      }
+    },
+	"deployPrefix": {
+      "type": "string",
+      "metadata": {
+        "description": "The prefix used for the nodes & dns. Must be lowercase. It should match with the following regular expression: ^[a-z][a-z0-9-]{1,61}[a-z0-9]$ or it will raise an error."
+      }
+    },
+	"vmSize": {
+	  "type": "string",
+	  "metadata": {
+	    "description": "The size of the virtual machine used for the deployment"
+	  },
+	  "allowedValues": [
+		"Standard_A0",
+		"Standard_A1",
+		"Standard_A2",
+		"Standard_A3",
+		"Standard_A4",
+		"Basic_A0",
+		"Basic_A1",
+		"Basic_A2",
+		"Basic_A3",
+		"Basic_A4",
+		"Standard_A5",
+		"Standard_A6",
+		"Standard_A7",
+		"Standard_D1",
+		"Standard_D2",
+		"Standard_D3",
+		"Standard_D4",
+		"Standard_D11",
+		"Standard_D12",
+		"Standard_D13",
+		"Standard_D14",
+		"Standard_D1_v2",
+		"Standard_D2_v2",
+		"Standard_D3_v2",
+		"Standard_D4_v2",
+		"Standard_D5_v2",
+		"Standard_D11_v2",
+		"Standard_D12_v2",
+		"Standard_D13_v2",
+		"Standard_D14_v2",
+		"Standard_DS1",
+		"Standard_DS2",
+		"Standard_DS3",
+		"Standard_DS4",
+		"Standard_DS11",
+		"Standard_DS12",
+		"Standard_DS13",
+		"Standard_DS14",
+		"Standard_G1",
+		"Standard_G2",
+		"Standard_G3",
+		"Standard_G4",
+		"Standard_G5",
+		"Standard_GS1",
+		"Standard_GS2",
+		"Standard_GS3",
+		"Standard_GS4",
+		"Standard_GS5",
+		"Standard_A8",
+		"Standard_A9",
+		"Standard_A10",
+		"Standard_A11"
+	  ],
+	  "defaultValue": "Standard_A2"
+	},
+	"deploymentType": {
+      "type": "string",
+      "defaultValue": "Nodes",
+      "allowedValues": [
+        "Nodes",
+        "Server"
```

Comment:
How about being able to create both in one deployment? Is that possible? 

For insipiration when kicking off deployments to multiple files based on user input, see this template https://github.com/Azure/azure-quickstart-templates/tree/master/create-hpc-cluster. Based on user input it can create headnode-rdma-disabled.json or headnode-rdma-enabled.json and also either linux or windows.